### DISCUSSION TITLE
Randomization of two variables

### DISCUSSION THREAD

**Discussion by andreaar5** [jspsych/jsPsych#2120] [Q&A]
Hi! 

Is it possible to randomize the order of two variables? 

In my experiment, I show two video stimuli and after, a question.
One video is the same for each trial and the other changes every time. 
In a timeline, can I randomize the order? 

Thanks in advantage, 
Andrea

**Comment by jodeleeuw** [comment]
Hi @andreaar5 ,

In the simplest case, which I'm not sure covers what you are describing, you can always just randomize the order in which you add trials to a timeline. For example:

```js

var timeline = [];

var trial_1 = { ... }
var trial_2 = { ... }

var random_order = jsPsych.randomization.shuffle([trial_1, trial_2]);

timeline.push(random_order[0], random_order[1]);
```

**Reply by andreaar5** [reply]
Thanks! 
the randomization works! 
the problem now is where to insert the _question_ variable. 
I paste the code: 

```html
<!DOCTYPE html>
<html>
    <head>
        <title>My experiment</title>
        <script src="jspsych.js"></script>
        <script src="plugins/jspsych-fullscreen.js"></script>
        <script src="plugins/jspsych-html-keyboard-response.js"></script>
        <script src="plugins/jspsych-html-button-response.js"></script>
	    <script src="plugins/jspsych-image-keyboard-response.js"></script>
	    <script src="plugins/jspsych-survey-multi-select.js" ></script>
        <script src="plugins/jspsych-preload.js" ></script>
        <script src="plugins/jspsych-survey-text.js"></script>
	    <script src="plugins/jspsych-instructions.js"></script>
	    <script src="plugins/jspsych-video-keyboard-response.js"></script>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
        <link rel="icon" href="data:;base64,iVBORw0KGgo=">
        <link href="css/jspsych.css" rel="stylesheet" type="text/css"></link>
        <style> 
            div {text-align: center; text-justify: inter-word;}
            p {margin-left: 50px; margin-right: 50px;}
            h2 {color: #333366;}
            h3 {color: #333366;}
            h4 {color: #333366;}
            video {height:95vh;
                object-fit: contain;}
            .jspsych-btn1 {
            background-color: #333366; color: white;
            font-size: 20px; font-family: 'Open Sans', 'Arial', sans-serif; 
            line-height: 1.4; display: inline-block;  border-radius: 4px;}
            .jspsych-btn2 {
            background-color: lightgray; color: black;
            font-size: 30px; font-family: 'Open Sans', 'Arial', sans-serif; 
            vertical-align: bottom; line-height: 1.5; display: inline-block;  border-radius: 4px;
            padding: 0px 12px; margin-top: 40px;; margin-left: 100px;}
            .jspsych-btn3 {
            background-color: lightgray; color: black;
            font-size: 30px; font-family: 'Open Sans', 'Arial', sans-serif; 
            vertical-align: bottom; line-height: 1.5; display: inline-block;  border-radius: 4px;
            padding: 0px 12px; margin-top: 40px; }
            </style>
        <body>  </body>
    </head>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script>
//giallo //#ffEb24; 
 /* create timeline */   
var timeline = [];

/*********** fullscreen mode ***********/
var fullscreen_mode = {
  type: 'fullscreen',
  message: "<p>Passare a modalità schermo intero</p>",
  button_label: "Continua",
  fullscreen_mode: true
};
//timeline.push(fullscreen_mode);

////// 1. PAGINA INIZIALE
var pagina_iniziale = {
	type: 'instructions',
	pages: ["<img src=immagini\\logo_orizzontale_COLORE.png width=300>"
        +"<h2>Benvenuto/a! </h2>" 
		+"Il presente studio ha lo scopo di valutare quanto l'età e l'esperienza motoria possano determinare la capacit&agrave di riconoscere un carico alzato durante una azione di vita quotidiana." 
		+"<br>"
        +"<footer><span style=font-size:12px>Non sar&aacute dato accesso ai dati a terze parti per indagini di mercato o pubblicit&aacute.</footer>",
        //spiegazione
        ],
		post_trial_gap: 0, 
	show_clickable_nav: true,
	};	
    timeline.push(pagina_iniziale);

var test_stimuli_C = [
{vid_c: 'VideoF\\0_F_CON.mp4', domanda:"", data:{ConditionName: '0_F_CON'}, nome: '0_F_CON', },   
{vid_c: 'VideoF\\1.5_F_CON.mp4', domanda:"", data:{ConditionName: '1.5_F_CON'}, nome: '1.5_F_CON', },  
{vid_c: 'VideoF\\3_F_CON.mp4', domanda:"", data:{ConditionName: '3_F_CON'}, nome: '3_F_CON', }, 
{vid_c: 'VideoF\\4.5_F_CON.mp4', domanda:"", data:{ConditionName: '4.5_F_CON'}, nome: '4.5_F_CON', },
{vid_c: 'VideoF\\6_F_CON.mp4', domanda:"", data:{ConditionName: '6_F_CON'}, nome: '6_F_CON', }, 
{vid_c: 'VideoF\\7.5_F_CON.mp4', domanda:"", data:{ConditionName: '7.5_F_CON'}, nome: '7.5_F_CON', },
{vid_c: 'VideoF\\9_F_CON.mp4', domanda:"", data:{ConditionName: '9_F_CON'}, nome: '9_F_CON', },  
];

var quest = {
        type: 'html-keyboard-response',
    	stimulus: "<span style=font-size:30px;>Nel <b>scatola</b> spostata nel secondo video era </span>", 
        choices: ['ArrowLeft','ArrowRight'],
        prompt: ['<button class="jspsych-btn3">più pesante <img src="immagini\\left.png"> </button>'
                + '<button class="jspsych-btn2"> <img src="immagini\\right.png"> più leggera</button>'] 
                + '<p>&nbsp;</p>'
                + "<span style=font-size:30px;>rispetto al primo.</span>", 
};

var quattro_con = {type: 'video-keyboard-response',
        stimulus: ['VideoF\\4.5_F_CON.mp4'], 
        autoplay: true, 
        trial_ends_after_video: true, response_allowed_while_playing: false, 
        choiches: jsPsych.NO_KEYS,
        post_trial_gap: 500,
        prompt: 'video_reference',
};

var video_con = {
type: "video-keyboard-response",
stimulus: [jsPsych.timelineVariable('vid_c'), 'VideoF\\4.5_F_CON.mp4'], 
randomize_order: true, 
autoplay: true, 
trial_ends_after_video: true,
response_allowed_while_playing: false, 
prompt: function () {return jsPsych.timelineVariable('nome', true)}, 
};

var exp_intro = {
type: 'html-keyboard-response',
stimulus: " ", choiches: jsPsych.ALL_KEYS,
prompt: "La prova &egrave finita. Ora inizia l'esperimento"
+ "<br>"
+"<br><p><u>Premi un tasto per continuare</p>",
}
timeline.push(exp_intro);


var exp_intro = {
type: 'html-keyboard-response',
stimulus: " ", choiches: jsPsych.ALL_KEYS,
prompt: "La prova &egrave finita. Ora inizia l'esperimento"
+ "<br>"
+"<br><p><u>Premi un tasto per continuare</p>",
}
timeline.push(exp_intro);


var random_order = jsPsych.randomization.shuffle([quattro_con, video_con]);

**var trial_video_concentric_condition = {
  timeline: [andom_order[0], random_order[1], quest ],
  timeline_variables: test_stimuli_C,
  randomize_order: true,
  sample: {
    type: 'fixed-repetitions',                  
    size: 1, // N repetitons of each trial_7 
},
       }
timeline.push(trial_video_concentric_condition);**


var exp_intro = {
type: 'html-keyboard-response',
stimulus: " ", choiches: jsPsych.ALL_KEYS,
prompt: "La prova &egrave finita. Ora inizia l'esperimento"
+ "<br>"
+"<br><p><u>Premi un tasto per continuare</p>",
}
timeline.push(exp_intro);

//////end
var end = {
	type: "html-keyboard-response",
		stimulus: " ",	 
    trial_duration: 3000,
    //response_allowed_while_playing: false,
	prompt: "<h2 style=color:red;> Grazie di aver partecipato all'esperimento!</h2>",
	};	
    timeline.push(end);



var today = new Date();
     var dd = String(today.getDate()).padStart(2, '0');
     var mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0!
     var yyyy = today.getFullYear();
     var ss = String(today.getSeconds()).padStart(2, '0');
     var min = String(today.getMinutes()).padStart(2, '0');
     var hh = String(today.getHours()).padStart(2, '0');
       today = mm +  dd + yyyy + '-' + hh + min + ss;


jsPsych.init({
	timeline: timeline,
        on_finish: function() {
            jsPsych.data.get().localSave('csv',today+'mydata.csv');
        }
    })
    </script>
</html>
```

so, it does not work very well, the order is randomized 
but it's the same every trial.

NB. this is a trial code

Thanks, 
Andrea
